### PR TITLE
JSON Editor unit tests

### PR DESCRIPTION
**Pull request by fiskus**
# Description

* Fixed existing config for `jest` unit testing
* Add unit tests for JSON Editor state

3 tests are failing and skipped. Code for these tests fixed in #1921

### CODE REVIEW

File: catalog/app/components/JsonEditor/State.spec.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+} from './State'
+
+import * as booleansNulls from './mocks/booleans-nulls'
+import * as compound from './mocks/compound'
+import * as deeplyNestedArray from './mocks/deeply-nested-array'
+import * as deeplyNestedObject from './mocks/deeply-nested-object'
+import * as incorrect from './mocks/incorrect'
+import * as regular from './mocks/regular'
+
+describe('Root keys for JSON editor', () => {
```
Comment (nl0) [inline catalog/app/components/JsonEditor/State.spec.js:15]: pls see a note on structuring test cases in a neighbour PR (#1922)

File: catalog/jest.config.js [hunk 2]
```diff
@@ -21,10 +21,7 @@ module.exports = {
     '.*\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
       '<rootDir>/internals/mocks/image.js',
   },
-  setupFilesAfterEnv: [
-    '<rootDir>/internals/testing/test-bundler.js',
-    '@testing-library/react/cleanup-after-each',
```
Comment (nl0) [inline catalog/jest.config.js:26]: why have you removed that? is it failing?

File: catalog/package.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
     "pretest": "npm run test:clean && npm run lint",
     "test:clean": "rimraf ./coverage",
     "test": "cross-env NODE_ENV=test jest --coverage",
-    "test:watch": "cross-env NODE_ENV=test jest --watchAll",
+    "test:watch": "cross-env NODE_ENV=test jest --watch",
```
Comment (nl0) [inline catalog/package.json:43]: why is this required?

**Review by nl0 — CHANGES_REQUESTED** [review]
pls structure the tests according to [this note](https://github.com/quiltdata/quilt/pull/1922#discussion_r526963142)

File: catalog/jest.config.js (hunk 2 above)
Comment (fiskus) [reply catalog/jest.config.js:26]: Yes, it's failing

And https://github.com/testing-library/react-testing-library/releases/tag/v9.0.0

File: catalog/app/components/JsonEditor/State.spec.js (hunk 1 above)
Comment (fiskus) [reply catalog/app/components/JsonEditor/State.spec.js:15]: It has this structure already. It can has some syntax or grammar mistakes, but the structure is `describe("Subject", it("verb + attributes"))`

File: catalog/package.json (hunk 3 above)
Comment (fiskus) [reply catalog/package.json:43]: It's not required. I don't see reasons, why we need `--watchAll` if `--watch` is enough
If there are reasons, please elaborate.

**Review by nl0 — CHANGES_REQUESTED** [review]
* [ ] name units of work more explicitly
* [ ] split complex test cases into simple ones

File: catalog/app/components/JsonEditor/State.spec.js [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+
+      expect(errors).toHaveLength(1)
+      expect(errors[0]).toBeInstanceOf(Error)
+      expect(errors[0].message).toMatch(/schema is invalid/)
+    })
+
+    describe('validator for Schema with enum types', () => {
+      const validate = validateOnSchema(regular.schema)
+
+      test("should return error, when value doesn't match enum", () => {
```
Comment (nl0) [inline catalog/app/components/JsonEditor/State.spec.js:142]: i believe these `test`s should be `it`s

**Review by nl0 — APPROVED** [review]
looks good, tho please replace `test`s w/ `it`s (i think they were renamed by mistake)
